src/Microsoft.TestPlatform.Common/ExtensionFramework/VSExtensionManager.cs

```diff
@@ -0,0 +1,203 @@
+        /// </summary>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate"), System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public IEnumerable<string> GetUnitTestExtensions()
+        {
+            try
+            {
+                return GetTestExtensionsInternal(Constants.UnitTestExtensionType);
+            }
+            catch (Exception ex)
+            {
+                string message = string.Format(CultureInfo.CurrentCulture, Resources.FailedToFindInstalledUnitTestExtensions, ex.Message);
+                throw new TestPlatformException(message, ex);
+            }
+        }
+
+        /// <summary>
+        /// Get the unit tests extensions
+        /// </summary>
+        private IEnumerable<string> GetTestExtensionsInternal(string extensionType)
+        {
+            IEnumerable<string> installedExtensions = new List<string>();
+
+            // Navigate to the IDE folder
+            this.vsInstallPath = new DirectoryInfo(typeof(ITestPlatform).GetTypeInfo().Assembly.GetAssemblyLocation()).Parent.Parent.Parent.FullName;
+            this.pathToDevenv = Path.Combine(vsInstallPath, DevenvExe);
+
+            // Check for devenv
+            if (!this.fileHelper.Exists(this.pathToDevenv))
+            {
+                throw new TestPlatformException(string.Format(CultureInfo.CurrentCulture, Resources.VSInstallationNotFound, this.vsInstallPath));
+            }
+
+            // Adding resolution paths for resolving dependencies.
+            var resolutionPaths = new List<string>() {
+                this.vsInstallPath,
+                Path.Combine(this.vsInstallPath, PrivateAssembliesDirName),
+                Path.Combine(this.vsInstallPath, PublicAssembliesDirName)
+            };
+            var assemblyResolver = new AssemblyResolver(resolutionPaths);
+
+            object extensionManager;
+            object settingsManager;
+                                       
+            settingsManager = SettingsManagerType.GetMethod("CreateForApplication", new Type[] { typeof(String) }).Invoke(null, new object[] { pathToDevenv });
+            if (settingsManager != null)
+            {
+                // create extension manager
+                extensionManager = Activator.CreateInstance(ExtensionManagerServiceType, settingsManager);
+
+                if (extensionManager != null)
+                {
+                    installedExtensions = ExtensionManagerServiceType.GetMethod("GetEnabledExtensionContentLocations", new Type[] { typeof(String) }).Invoke(
+                                               extensionManager, new object[] { extensionType }) as IEnumerable<string>;
+                }
+            
+                // Dispose the settings manager
+                IDisposable disposable = (settingsManager as IDisposable);
```

Comment:
Is this because of reflection? If this is a mandatory step then do it under a finally